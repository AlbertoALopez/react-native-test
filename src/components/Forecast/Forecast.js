/*
 *  React component that contains all the forecast rendering
 */

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';


const styles = StyleSheet.create({
  bigText: {
    // flex: 2,
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
    color: '#FFF',
  },
  mainText: {
    // flex: 1,
    fontSize: 16,
    textAlign: 'center',
    color: '#FFF',
  },
});

export default class Forecast extends Component {
  static propTypes = {
    main: React.PropTypes.string.isRequired,
    description: React.PropTypes.string.isRequired,
    temp: React.PropTypes.number.isRequired,
  }

  render() {
    return (
      <View>
        <Text style={styles.bigText}>
          {this.props.main}
        </Text>
        <Text style={styles.mainText}>
          Current conditions: {this.props.description}
        </Text>
        <Text style={styles.bigText}>
          {this.props.temp} C
        </Text>
      </View>
    );
  }
}
