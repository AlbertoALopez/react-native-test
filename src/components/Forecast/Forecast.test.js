import 'react-native';
import React from 'react';
import Forecast from './Forecast.js';
import renderer from 'react-test-renderer';


it('Forecast renders correctly', () => {
  const tree = renderer.create(
    <Forecast
      main="test"
      description="test"
      temp={24.5}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
