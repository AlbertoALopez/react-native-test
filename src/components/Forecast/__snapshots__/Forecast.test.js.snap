exports[`test Forecast renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "flex": 2,
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "flex": 1,
        "fontSize": 20,
        "textAlign": "center",
      }
    }>
    Current conditions: 
    test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "flex": 2,
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    24.5
  </Text>
</View>
`;
